### PR TITLE
Implement unit tests in github actions

### PR DESCRIPTION
**Pull request by PietrH**
This workflow will install Python dependencies, run tests and lint with a variety of Python versions

 For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python

## Todo

- [ ] Remove Travis CI
- [x] Check test coverage, currently test_reporting.py is an empty file
- [x] consider automatic publishing of releases to PyPI

### CODE REVIEW

**Comment by PietrH** [comment]
Discussed with Peter, currently automatic publishing of github releases to pypi is not a priority. Existing webhook to do this seems broken. Will be removed.

**Comment by PietrH** [comment]
Removing Travis CI support postponed for now

**Comment by PietrH** [comment]
Currently coverage is rather low, only at 63%
```
---------- coverage: platform linux, python 3.10.7-final-0 -----------
Name                   Stmts   Miss  Cover
------------------------------------------
pywhip/__init__.py         6      0   100%
pywhip/cli.py             33     20    39%
pywhip/pywhip.py         164    126    23%
pywhip/reporters.py       33     20    39%
pywhip/validators.py     242     13    95%
------------------------------------------
TOTAL                    478    179    63%
```

We can consider increasing this coverage in a future pull request

**Comment by PietrH** [comment]
Ready for review

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python package
+
+on:
+  push:
+    branches: [ "master" ]
```
Comment (peterdesmet) [inline .github/workflows/python-package.yml:8]: Can you replace this with:

```
- main
- master
```

Same for line 10.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -101,7 +101,7 @@ Before you submit a pull request, check that it meets these guidelines:
 2. If the pull request adds functionality, the docs should be updated. Put
    your new functionality into a function with a docstring, and add the
    feature to the list in README.rst.
-3. The pull request should work for Python 2.7, 3.3, 3.4 and 3.5. Check
+3. The pull request should work for Python 2.7, 3.3, 3.5, 3.7 and 3.8. Check
    https://travis-ci.org/inbo/pywhip/pull_requests
```
Comment (peterdesmet) [inline CONTRIBUTING.rst:105]: Update this with GitHub Actions link

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: 'coverage'
+on:
+    pull_request:
```
Comment (peterdesmet) [inline .github/workflows/coverage.yml:3]: I think we can run coverage for regular pushes as well, so would updated to:

```
on:
  push:
    branches:
      - main
      - master
  pull_request:
    branches:
      - main
      - master
```

File: .github/workflows/coverage.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        branches:
+            - master
+            - main
+jobs:
+    coverage:
```
Comment (peterdesmet) [inline .github/workflows/coverage.yml:8]: This file uses 4 space indents, the other one 2. Would use 2 for both.

**Review by peterdesmet — CHANGES_REQUESTED** [review]
Nice work! Some minor (mainly) stylistic changes suggested.

**Comment by PietrH** [comment]
Ok by me to delete the branch

**Pull request merged**